packages/gatsby-plugin-mdx/README.md

```diff
@@ -145,192 +109,13 @@ module.exports = {
-
-You can also set `options.defaultLayouts.default` if you only want to
-use one layout for all MDX pages that don't already have a layout defined.
-
-```js
-module.exports = {
-  siteMetadata: {
-    title: `Gatsby MDX Kitchen Sink`,
-  },
-  plugins: [
-    {
-      resolve: `gatsby-plugin-mdx`,
-      options: {
-        defaultLayouts: {
-          posts: require.resolve("./src/components/posts-layout.js"),
-          default: require.resolve("./src/components/default-page-layout.js"),
-        },
-      },
-    },
-    {
-      resolve: `gatsby-source-filesystem`,
-      options: {
-        name: `posts`,
-        path: `${__dirname}/src/posts/`,
-      },
-    },
-    {
-      resolve: "gatsby-plugin-page-creator",
-      options: {
-        path: `${__dirname}/src/posts`,
-      },
-    },
-  ],
-}
-```
-
-#### Imports
-
-When importing a react component into your MDX, you can import it using the `import`
-statement as in JavaScript.
-
-```mdx
-import { SketchPicker } from "react-color"
-
-# Hello, world!
-
-Here's a color picker!
-
-<SketchPicker />
-```
-
-_**Note:** You should rerun your Gatsby development environment to update imports in MDX files. Otherwise, you'll get a `ReferenceError` for new imports. You can use the shortcodes approach if that is an issue for you._
-
-#### Shortcodes
-
-If you want to allow usage of a component from anywhere (often referred to as a
-shortcode), you can pass it to the
-[MDXProvider](https://www.gatsbyjs.com/docs/mdx/customizing-components/).
-
-```js
-// src/components/layout.js
-import React from "react"
-import { MDXProvider } from "@mdx-js/react"
-import { Link } from "gatsby"
-import { YouTube, Twitter, TomatoBox } from "./ui"
-
-const shortcodes = { Link, YouTube, Twitter, TomatoBox }
-
-export default ({ children }) => (
-  <MDXProvider components={shortcodes}>{children}</MDXProvider>
-)
-```
-
-Then, in any MDX file, you can navigate using `Link` and render `YouTube`, `Twitter`, and `TomatoBox` components without
-an import.
-
-```mdx
-# Hello, world!
-
-Here's a YouTube embed
-
-<TomatoBox>
-  <YouTube id="123abc" />
-</TomatoBox>
-```
+#### `gatsby-remark-*` plugins
 
-[Read more about MDX shortcodes](https://mdxjs.com/blog/shortcodes)
+This config option is used for compatibility with a set of plugins many people [use with remark](https://www.gatsbyjs.com/plugins/?=gatsby-remark-) that require the gatsby environment to function properly. In some cases, like [gatsby-remark-prismjs](https://www.gatsbyjs.com/plugins/gatsby-remark-prismjs/), it makes more sense to use a library like [prism-react-renderer](https://github.com/FormidableLabs/prism-react-renderer) to render codeblocks using a [React component](/api-reference/mdx-provider). In other cases, like [gatsby-remark-images](https://www.gatsbyjs.com/plugins/gatsby-remark-images/), the interaction with the Gatsby APIs is well deserved because the images can be optimized by Gatsby and you should continue using it.
```

Comment:
```suggestion
This config option is used for compatibility with a set of plugins many people [use with remark](https://www.gatsbyjs.com/plugins/?=gatsby-remark-) that require the Gatsby environment to function properly. In some cases, like [gatsby-remark-prismjs](https://www.gatsbyjs.com/plugins/gatsby-remark-prismjs/), it makes more sense to use a library like [prism-react-renderer](https://github.com/FormidableLabs/prism-react-renderer) to render codeblocks using a [React component](/api-reference/mdx-provider). In other cases, like [gatsby-remark-images](https://www.gatsbyjs.com/plugins/gatsby-remark-images/), the interaction with the Gatsby APIs is well deserved because the images can be optimized by Gatsby and you should continue using it.
```